platforms/software/dependency-management/src/main/java/org/gradle/internal/component/ResolutionFailureHandler.java

```diff
@@ -216,6 +225,19 @@ private <FAILURE extends ResolutionFailure> AbstractResolutionFailureException d
             .filter(describer -> describer.canDescribeFailure(failure))
             .findFirst()
             .map(describer -> describer.describeFailure(failure, schema))
+            .map(this::reportExceptionAsProblem)
```

Comment:
We cannot use `internalReporter.throwing()` because of this, right?

Reply:
Right, the `ResolutionFailureHandler` isn't actually throwing the exception yet, the dependency resolution code handles that.  This class is merely responsible for creating the exception that will be thrown.